aicsimageio/metadata/utils.py

```diff
@@ -0,0 +1,304 @@
+KNOWN_INVALID_OME_XSD_REFERENCES = [
+    "www.openmicroscopy.org/Schemas/ome/2013-06",
+    "www.openmicroscopy.org/Schemas/OME/2012-03",
+]
+REPLACEMENT_OME_XSD_REFERENCE = "www.openmicroscopy.org/Schemas/OME/2016-06"
+
+###############################################################################
+
+
+def generate_ome_image_id(image_id: Union[str, int]) -> str:
+    """
+    Naively generates the standard OME image ID using a provided ID.
+
+    Parameters
+    ----------
+    image_id: Union[str, int]
+        A string or int representating the ID for an image.
+        In the context of the usage of this function, this is usually used with the
+        index of the scene / image.
+
+    Returns
+    -------
+    ome_image_id: str
+        The OME standard for image IDs.
+    """
+    return f"Image:{image_id}"
+
+
+def clean_ome_xml_for_known_issues(xml: str) -> str:
+    """
+    Clean an OME XML string for known issues created by AICS systems and tools.
+
+    Commonly this is used for cleaning a file produced by AICS prior to noticing the
+    issue (2021), or for other users of aicsimageio as a whole prior to 4.x series of
+    releases.
+
+    The result of this function should be an OME XML string that is relatively the
+    same (no major pieces missing) but that validates against the references OME
+    XSD.
+
+    Parameters
+    ----------
+    xml: str
+        The OME XML string to clean for errors.
+
+    Returns
+    -------
+    cleaned_xml: str
+        The cleaned OME XML string.
+    """
+    # Store list of changes to print out with warning
+    metadata_changes = []
+
+    # Fix xsd reference
+    # This is from OMEXML object just having invalid reference
+    for known_invalid_ref in KNOWN_INVALID_OME_XSD_REFERENCES:
+        if known_invalid_ref in xml:
+            xml = xml.replace(
+                known_invalid_ref,
+                REPLACEMENT_OME_XSD_REFERENCE,
+            )
+            metadata_changes.append(
+                f"Replaced '{known_invalid_ref}' with "
+                f"'{REPLACEMENT_OME_XSD_REFERENCE}'."
+            )
+
+    # Read in XML
+    root = ET.fromstring(xml)
+
+    # Get the namespace
+    # In XML etree this looks like
+    # "{http://www.openmicroscopy.org/Schemas/OME/2016-06}"
+    # and must prepend any etree finds
+    namespace_matches = re.match(r"\{.*\}", root.tag)
+    if namespace_matches is not None:
+        namespace = namespace_matches.group(0)
+    else:
+        raise ValueError("XML does not contain a namespace")
+
+    # Find all Image elements and fix IDs
+    # This is for certain for test files of o.urs and ACTK files
+    for image_index, image in enumerate(root.findall(f"{namespace}Image")):
+        image_id = image.get("ID")
+        if not image_id.startswith("Image"):
+            image.set("ID", generate_ome_image_id(image_id))
+            metadata_changes.append(
+                f"Updated attribute 'ID' from '{image_id}' to 'Image:{image_id}' "
```

Comment:
could use `generate_ome_image_id` anywhere we see Image:{image_id} (reuse the return value from the invocation two lines up?)

Reply:
Yep!